Formula/lilypond.rb

```diff
@@ -0,0 +1,96 @@
+class Lilypond < Formula
+  desc "Music notation for everyone"
+  homepage "https://lilypond.org"
+  url "https://lilypond.org/download/sources/v2.22/lilypond-2.22.1.tar.gz"
+  sha256 "72ac2d54c310c3141c0b782d4e0bef9002d5519cf46632759b1f03ef6969cc30"
+  license "GPL-3.0-or-later"
+
+  depends_on "bdw-gc" => :build
+  depends_on "bison" => :build # Lilypond requires bison 2.4.1 or above
+  depends_on "fontforge" => :build
+  depends_on "gettext" => :build
+  depends_on "glib" => :build
+  depends_on "gmp" => :build
+  depends_on "imagemagick" => :build
+  depends_on "libtool" => :build
+  depends_on "libunistring" => :build
+  depends_on "perl" => :build
+  depends_on "pkg-config" => :build
+  depends_on "readline" => :build
+  depends_on "t1utils" => :build
+  depends_on "texi2html" => :build
+  depends_on "texinfo" => :build # lilypond requires texinfo 6.1 or above
+  depends_on "texlive" => :build
+
+  depends_on "fontconfig"
+  depends_on "freetype"
+  depends_on "ghostscript"
+  depends_on "pango"
+  depends_on "python@3.9"
+
+  uses_from_macos "flex" => :build
+  uses_from_macos "libffi" => :build
+
+  resource "guile@1" do
+    url "https://ftp.gnu.org/gnu/guile/guile-1.8.8.tar.gz"
+    sha256 "c3471fed2e72e5b04ad133bbaaf16369e8360283679bcf19800bc1b381024050"
+  end
+
+  resource "urw-fonts" do
+    url "https://github.com/ArtifexSoftware/urw-base35-fonts/archive/20200910.tar.gz"
+    sha256 "e0d9b7f11885fdfdc4987f06b2aa0565ad2a4af52b22e5ebf79e1a98abd0ae2f"
+  end
+
+  def install
+    # Install guile 1.8.8 and distribute with lilypond
+    resource("guile@1").stage buildpath/"guile1"
+    cd "#{buildpath}/guile1" do
+      system "./configure", "--disable-dependency-tracking",
+                            "--prefix=#{libexec}",
+                            "--with-libgmp-prefix=#{Formula["gmp"].opt_prefix}",
+                            "--with-libreadline-prefix=#{Formula["readline"].opt_prefix}"
+      system "make", "install"
+    end
+    Pathname.glob(libexec/"lib/*.dylib") do |dylib|
+      (libexec/"lib").install_symlink dylib.basename => "#{dylib.basename(".dylib")}.so"
+    end
+    ENV.append_path "PATH", libexec/"bin"
+    ENV.append_path "PKG_CONFIG_PATH", libexec/"lib/pkgconfig"
+
+    # needed to solve conflicts with emacs
+    inreplace "config.make.in",
+      %r{^\s*elispdir\s*=\s*\$\(datadir\)/emacs/site-lisp\s*$},
+      "elispdir = $(datadir)/emacs/site-lisp/lilypond"
+
+    resource("urw-fonts").stage buildpath/"urw"
+
+    system "./configure",
+            "--prefix=#{prefix}",
+            "--with-texgyre-dir=#{Formula["texlive"].opt_libexec}/share/texmf-dist/fonts/opentype/public/tex-gyre",
```

Comment:
We will probably change this path depending on what will be decided about using `libexec` in #83738